Vostok.Commons.Collections/ConcurrentBoundedQueue.cs

```diff
@@ -127,29 +138,52 @@ public int Drain(T[] buffer, int index, int count)
         /// <summary>
         /// Asynchronously waits until something is available to <see cref="Drain"/>.
         /// </summary>
-        public Task WaitForNewItemsAsync() => canDrain.Task;
+        public Task WaitForNewItemsAsync() => canDrainAny.Task;
 
         /// <summary>
         /// Asynchronously waits until something is available to <see cref="Drain"/> or the provided <paramref name="timeout"/> expires.
         /// </summary>
         /// <returns><c>true</c> if there is something to drain, <c>false</c> otherwise.</returns>
         public async Task<bool> TryWaitForNewItemsAsync(TimeSpan timeout)
         {
-            if (canDrain.Task.IsCompleted)
+            if (canDrainAny.Task.IsCompleted)
                 return true;
 
             using (var cts = new CancellationTokenSource())
             {
-                var delay = Task.Delay(timeout, cts.Token);
+                var waitTimeout = Task.Delay(timeout, cts.Token);
 
-                var result = await Task.WhenAny(canDrain.Task, delay).ConfigureAwait(false);
-                if (result == delay)
+                var result = await Task.WhenAny(canDrainAny.Task, waitTimeout).ConfigureAwait(false);
+                if (result == waitTimeout)
```

Comment:
I suggest to make it homogenic to the batch method: we should probably remove this `if` here and return `canDrainAny.Task.IsCompleted` instead.

Reply:
We don't need to cancel token if `result == waitTimeout`.